injector/src/test/java/edu/ucr/cs/riple/injector/TypeUseAnnotationTest.java

```diff
@@ -0,0 +1,244 @@
+                "custom.example.Untainted"),
+            new AddTypeUseMarkerAnnotation(
+                new OnParameter("Foo.java", "test.Foo", "baz(java.lang.Object)", 0),
+                "custom.example.Untainted"))
+        .start();
+  }
+
+  @Test
+  public void deletionOnFullyQualifiedTypeNamesTest() {
+    injectorTestHelper
+        .addInput(
+            "Foo.java",
+            "package test;",
+            "import custom.example.Untainted;",
+            "public class Foo {",
+            "   java.lang.@Untainted Object bar;",
+            "   java.util.@Untainted Map<java.lang.@Untainted String, @Untainted String[]> f0;",
+            "   java.lang.@Untainted Object baz(java.lang.@Untainted Object param) {;",
+            "       java.lang.@Untainted Object localVar;",
+            "       return new Object();",
+            "   }",
+            "}")
+        .expectOutput(
+            "package test;",
+            "import custom.example.Untainted;",
+            "public class Foo {",
+            "   java.lang.Object bar;",
+            "   java.util.Map<java.lang.String, String[]> f0;",
+            "   java.lang.Object baz(java.lang.Object param) {;",
+            "       java.lang.Object localVar;",
+            "       return new Object();",
+            "   }",
+            "}")
+        .addChanges(
+            new RemoveTypeUseMarkerAnnotation(
+                new OnField("Foo.java", "test.Foo", Set.of("bar")), "custom.example.Untainted"),
+            new RemoveTypeUseMarkerAnnotation(
+                new OnField("Foo.java", "test.Foo", Set.of("f0")), "custom.example.Untainted"),
+            new RemoveTypeUseMarkerAnnotation(
+                new OnMethod("Foo.java", "test.Foo", "baz(java.lang.Object)"),
+                "custom.example.Untainted"),
+            new RemoveTypeUseMarkerAnnotation(
+                new OnLocalVariable("Foo.java", "test.Foo", "baz(java.lang.Object)", "localVar"),
+                "custom.example.Untainted"),
+            new RemoveTypeUseMarkerAnnotation(
+                new OnParameter("Foo.java", "test.Foo", "baz(java.lang.Object)", 0),
+                "custom.example.Untainted"))
+        .start();
+  }
+
+  @Test
+  public void additionOnArrayTest() {
+    injectorTestHelper
+        .addInput(
+            "Foo.java",
+            "package test;",
+            "public class Foo<T> {",
+            "   public void foo() {",
+            "      java.util.Map<java.lang.String, String[]> f0;",
+            "      T[] f1;",
+            "      String[] f2;",
+            "   }",
+            "}")
+        .expectOutput(
+            "package test;",
+            "import edu.ucr.UnTainted;",
+            "public class Foo<T> {",
+            "   public void foo() {",
+            "      java.util.@UnTainted Map<java.lang.@UnTainted String, @UnTainted String[]> f0;",
+            "      @UnTainted T[] f1;",
```

Comment:
So right now we do not support injecting `@UnTainted T @UnTainted []`; we only support annotating the elements of the array (the second annotation applies to the array pointer itself).  Is this correct?

Reply:
Yes, we do not add the second annotation. Please let me know if you would like to add that as well.

Reply:
At the least we need to document what we do. If we want to inject nullability type use annotations, we will need support for both. Even for tainting, if we ever have a case where the array itself may be tainted (not its contents) we will need to support both locations. But up to you on whether to do it in this PR

Reply:
I thought about this, and I think maybe we should update this in a followup PR, as it may have changes in type declaration annotations as well. I think for now, even if we instantiate a `AddMarkerAnnotation` and not a `AddTypeUseMarkerAnnotation` annotation, for arrays we annotate as below:
```java
@Annot T[]
```

Which I think `AddMarkerAnnotation` should annotate as ```T @Annot []```
Added a comment / todo to update.  669b806be6f31ddcfdc584c666bfd014e1d9c131